libc/config/linux/api.td

```diff
@@ -175,6 +175,8 @@ def PThreadAPI : PublicAPI<"pthread.h"> {
       "__pthread_start_t",
       "__pthread_tss_dtor_t",
       "pthread_attr_t",
+      "pthread_cond_t",
+      "pthread_condattr_t",
```

Comment:
It might be nice to land https://github.com/llvm/llvm-project/pull/88987 then rebase this PR on top of that. Marking this as draft for now.